src/ceph-daemon/ceph-daemon

```diff
@@ -1149,10 +1161,13 @@ def command_bootstrap():
                 tmp_config.name: '/etc/ceph/ceph.conf:z',
             },
         )
-        out, err, ret = call(c.run_cmd(), c.entrypoint)
+        out, err, ret = call(c.run_cmd(), desc=c.entrypoint, timeout=30)
         if ret == 0:
             break
-        logger.info('mon is still not available yet, waiting...')
+        elif retry_num > retry_max:
```

Comment:
Can we encapsulate this into a function or context manager? I'm seeing this loop 3 times in this PR

```python
def retry(what, retry_max, func):
        logger.info('Waiting for {what}...')
        retry_num = 1
        while True:
            if func()
                break
            elif retry_num > retry_max:
                raise Error('{what} is not available after %s tries' % retry_max)
            logger.info('{what} not yet available, waiting (%s/%s)...' % (retry_num, retry_max))
            retry_num += 1
            time.sleep(1)
```
?

Reply:
Thanks! Refactored this into an `is_available` function here: https://github.com/ceph/ceph/pull/32049/commits/00028bd40ffde9ccba6d607f629dca6ef94b9f98#diff-8b586ec9c3ad3e8421a8858888f7ddf0R251-R279